sdk/messaging/azservicebus/admin_client_queue.go

```diff
@@ -5,33 +5,340 @@ package azservicebus
+	// are forwarded to.
+	ForwardTo *string
+
+	// ForwardDeadLetteredMessagesTo is the absolute URI of the entity to forward dead letter messages
+	ForwardDeadLetteredMessagesTo *string
+}
+
+// QueueRuntimeProperties represent dynamic properties of a queue, such as the ActiveMessageCount.
+type QueueRuntimeProperties struct {
+	// Name is the name of the queue.
+	Name string
+
+	// SizeInBytes - The size of the queue, in bytes.
+	SizeInBytes int64
+
+	// CreatedAt is when the entity was created.
+	CreatedAt time.Time
+
+	// UpdatedAt is when the entity was last updated.
+	UpdatedAt time.Time
+
+	// AccessedAt is when the entity was last updated.
+	AccessedAt time.Time
+
+	// TotalMessageCount is the number of messages in the queue.
+	TotalMessageCount int64
+
+	// ActiveMessageCount is the number of active messages in the entity.
+	ActiveMessageCount int32
+
+	// DeadLetterMessageCount is the number of dead-lettered messages in the entity.
+	DeadLetterMessageCount int32
+
+	// ScheduledMessageCount is the number of messages that are scheduled to be enqueued.
+	ScheduledMessageCount int32
+
+	// TransferDeadLetterMessageCount is the number of messages transfer-messages which are dead-lettered
+	// into transfer-dead-letter subqueue.
+	TransferDeadLetterMessageCount int32
+
+	// TransferMessageCount is the number of messages which are yet to be transferred/forwarded to destination entity.
+	TransferMessageCount int32
+}
+
+type AddQueueResponse struct {
+	// Value is the result of the request.
+	Value *QueueProperties
+	// RawResponse is the *http.Response for the request.
+	RawResponse *http.Response
+}
+
+type AddQueueWithPropertiesResponse struct {
+	// Value is the result of the request.
+	Value *QueueProperties
+	// RawResponse is the *http.Response for the request.
+	RawResponse *http.Response
+}
+
+type GetQueueResponse struct {
+	// Value is the result of the request.
+	Value *QueueProperties
+	// RawResponse is the *http.Response for the request.
+	RawResponse *http.Response
+}
+
+type UpdateQueueResponse struct {
+	// Value is the result of the request.
+	Value *QueueProperties
+	// RawResponse is the *http.Response for the request.
+	RawResponse *http.Response
+}
+
+type GetQueueRuntimePropertiesResponse struct {
+	// Value is the result of the request.
+	Value *QueueRuntimeProperties
+	// RawResponse is the *http.Response for the request.
+	RawResponse *http.Response
+}
+
+type DeleteQueueResponse struct {
+	RawResponse *http.Response
+}
+
+// AddQueue creates a queue using defaults for all options.
+func (ac *AdminClient) AddQueue(ctx context.Context, queueName string) (*AddQueueResponse, error) {
+	return ac.AddQueueWithProperties(ctx, &QueueProperties{
+		Name: queueName,
+	})
+}
+
+// CreateQueue creates a queue with configurable properties.
+func (ac *AdminClient) AddQueueWithProperties(ctx context.Context, properties *QueueProperties) (*AddQueueResponse, error) {
```

Comment:
I had a discussion about a similar thing with @jhendrixMSFT about this before: [(link)](https://github.com/Azure/azure-event-hubs-go/pull/226#discussion_r671504002)

The gist was that it's okay if the nil passes through and it panics _but_ the critical part is it needs to be close to the point of origin. This one should fit that bill since we panic immediately because we do actually try to use properties.

So I'm going to leave this one with the acceptance that this is the Go way (even though it was counter to my own instincts at the time).